smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -179,35 +188,36 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         require(sigs.length == claims.length, "INV_CLM_LEN");
         require(sigs.length == signatureData.length, "INV_SIG_LEN");
 
+        uint256 intermediateNonce = lastNonceSubmitted;
+        require(intermediateNonce + 1 == claims[0].nonce, "INV_ORD");
+
         for (uint256 i = 0; i < sigs.length; i++) {
+            intermediateNonce++;
             _submitProphecyClaimAggregatedSigs(sigs[i], claims[i], signatureData[i]);
         }
+
+        lastNonceSubmitted = intermediateNonce;
     }
 
     function submitProphecyClaimAggregatedSigs(
         bytes32 hashDigest,
         ClaimData calldata claimData,
         SignatureData[] calldata signatureData
     ) external {
+        uint256 previousNonce = lastNonceSubmitted;
+        require(previousNonce + 1 == claimData.nonce, "INV_ORD");
+
```

Comment:
As discussed on call, let's move line 213 to 209 before the call to `_submitProphecyClaimAggregatedSigs `.